src/main/java/org/jabref/logic/xmp/DublinCoreExtractor.java

```diff
@@ -0,0 +1,387 @@
+package org.jabref.logic.xmp;
+
+import java.io.IOException;
+import java.util.Calendar;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.Map.Entry;
+
+import org.jabref.logic.TypedBibEntry;
+import org.jabref.model.database.BibDatabaseMode;
+import org.jabref.model.entry.Author;
+import org.jabref.model.entry.AuthorList;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.FieldName;
+import org.jabref.model.entry.Month;
+
+import org.apache.xmpbox.DateConverter;
+import org.apache.xmpbox.schema.DublinCoreSchema;
+
+public class DublinCoreExtractor {
+
+    private final DublinCoreSchema dcSchema;
+    private final XMPPreferences xmpPreferences;
+
+    private final BibEntry bibEntry;
+
+    public DublinCoreExtractor(DublinCoreSchema dcSchema, XMPPreferences xmpPreferences, BibEntry resolvedEntry) {
+        this.dcSchema = dcSchema;
+        this.xmpPreferences = xmpPreferences;
+
+        this.bibEntry = resolvedEntry;
+    }
+
+    /**
+     * Editor in BibTex - Contributor in DublinCore
+     *
+     * @param bibEntry The BibEntry object, which is filled during metadata extraction.
+     * @param dcSchema Metadata in DublinCore format.
+     */
+    private void extractEditor() {
+        List<String> contributors = dcSchema.getContributors();
+        if ((contributors != null) && !contributors.isEmpty()) {
+            bibEntry.setField(FieldName.EDITOR, String.join(" and ", contributors));
+        }
+    }
+
+    /**
+     * Author in BibTex - Creator in DublinCore
+     *
+     * @param bibEntry The BibEntry object, which is filled during metadata extraction.
+     * @param dcSchema Metadata in DublinCore format.
+     */
+    private void extractAuthor() {
+        List<String> creators = dcSchema.getCreators();
+        if ((creators != null) && !creators.isEmpty()) {
```

Comment:
I think we have a method in our own StringUtil class which combines both checks: IsNullOREmpty

Reply:
Done with https://github.com/JabRef/jabref/pull/3710/commits/366aeed6a6870386c9eb4ba657332faff8f7d576. Please check the conditions again. It's not that natural to negate the isNullOrEmpty... Maybe another function would help in the StringUtils.